server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java

```diff
@@ -4407,14 +4407,35 @@ public Vlan createVlanAndPublicIpRange(final CreateVlanIpRangeCmd cmd) throws In
         String endIP = cmd.getEndIp();
         final String newVlanGateway = cmd.getGateway();
         final String newVlanNetmask = cmd.getNetmask();
+        Long networkId = cmd.getNetworkID();
+        Long physicalNetworkId = cmd.getPhysicalNetworkId();
+
+        // Verify that network exists
+        Network network = null;
+        if (networkId != null) {
+            network = _networkDao.findById(networkId);
+            if (network == null) {
+                throw new InvalidParameterValueException("Unable to find network by id " + networkId);
+            } else {
+                zoneId = network.getDataCenterId();
+                physicalNetworkId = network.getPhysicalNetworkId();
+            }
+        }
+
         String vlanId = cmd.getVlan();
+        if (StringUtils.isBlank(vlanId)) {
+            vlanId = Vlan.UNTAGGED;
+            if (network != null & network.getTrafficType() == TrafficType.Guest) {
+                boolean connectivityWithoutVlan = isConnectivityWithoutVlan(network);
+                vlanId = getNetworkVlanId(network, connectivityWithoutVlan);
+            }
+        }
+
         // TODO decide if we should be forgiving or demand a valid and complete URI
         if (!(vlanId == null || "".equals(vlanId) || vlanId.startsWith(BroadcastDomainType.Vlan.scheme()))) {
             vlanId = BroadcastDomainType.Vlan.toUri(vlanId).toString();
         }
         final Boolean forVirtualNetwork = cmd.isForVirtualNetwork();
```

Comment:
does this need to be cleaned up ?

Reply:
no @weizhouapache, vlanId from the network can be empty/null.